java/src/test/java/com/spotify/voyager/jni/IndexTest.java

```diff
@@ -239,6 +239,27 @@ private void runTestWith(
           }
         }
       }
+      // Test markDeleted and unmarkDeleted
+      try {
+        index.markDeleted(0);
+        assert (true);
+      } catch (Exception e) {
+        assert (false);
+      }
+      // If we call it again it should trigger an exception
+      try {
+        index.markDeleted(0);
+        assert (false);
+      } catch (Exception e) {
+        assert (true);
+      }
+      // Now let's unmark it
+      try {
+        index.unmarkDeleted(0);
+        assert (true);
+      } catch (Exception e) {
+        assert (false);
+      }
```

Comment:
Could you update this test to not rely on exceptions as flow control and to use JUnit assertions? Would be better to mark the item as deleted, try to retrieve it, and assert that it is not returned. Similar for the `unmarkDeleted` case.

Reply:
Sure.